File: benchmarks/compute/sum_squares.go
// Sum of squares benchmark
// Build: go build -o sum_squares_go sum_squares.go

package main

import (
	"fmt"
	"os"
)

func sumSquares(n int64) int64 {
	var acc int64 = 0
	for i := int64(1); i <= n; i++ {
		acc += i * i
	}
	return acc
}

func main() {
	result := sumSquares(1_000_000)
	fmt.Println(result)

	// Expected: 333333833333500000
	if result == 333333833333500000 {
		os.Exit(0)
	} else {
		os.Exit(1)
	}
}

File: benchmarks/compute/sum_squares.rs
// Sum of squares benchmark
// Compile: rustc -O -o sum_squares_rust sum_squares.rs
//
// Note: n=1M is safe for i64. Limits above ~3M risk overflow.

fn sum_squares(n: i64) -> i64 {
    let mut acc: i64 = 0;
    for i in 1..=n {
        acc += i * i;
    }
    acc
}

fn main() {
    let result = sum_squares(1_000_000);
    println!("{}", result);

    // Expected: 333333833333500000
    std::process::exit(if result == 333333833333500000 { 0 } else { 1 });
}
